homeassistant/components/media_player/sonos.py

```diff
@@ -53,13 +55,18 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     if not players:
         players = soco.discover(interface_addr=config.get('interface_addr',
                                                           None))
-
     if not players:
         _LOGGER.warning('No Sonos speakers found.')
         return False
 
-    add_devices(SonosDevice(hass, p) for p in players)
-    _LOGGER.info('Added %s Sonos speakers', len(players))
+    if group_devices:
+        # Only care about whole groups
+        groups = {player.group.uid: player.group for player in players}
```

Comment:
See my other answer, in my experiments a lonely player has a group attached to it where the player is the coordinator so it seems there are always groups.